algo/src/main/java/org/neo4j/graphalgo/JaccardProc.java

```diff
@@ -33,24 +54,161 @@
             "YIELD source1, source2, count1, count2, intersection, jaccard - computes jaccard similarities")
-    public Stream<JaccardResult> jaccard(
+    public Stream<SimilarityResult> jaccardStream(
             @Name(value = "data", defaultValue = "null") List<Map<String,Object>> data,
             @Name(value = "config", defaultValue = "{}") Map<String, Object> config) {
+        ProcedureConfiguration configuration = ProcedureConfiguration.create(config);
+        double similarityCutoff = configuration.get("similarityCutoff", -1D);
+        long degreeCutoff = configuration.get("degreeCutoff", 0L);
 
-        double similiarityCutoff = ((Number) config.getOrDefault("similarityCutoff", -1D)).doubleValue();
-        long degreeCutoff = ((Number) config.getOrDefault("degreeCutoff", 0L)).longValue();
         InputData[] ids = fillIds(data, degreeCutoff);
         int length = ids.length;
+        TerminationFlag terminationFlag = TerminationFlag.wrap(transaction);
+
+        int concurrency = configuration.getConcurrency();
+
+        return jaccardStreamMe(ids, length, terminationFlag, concurrency, similarityCutoff);
+    }
+
+    @Procedure(name = "algo.jaccard", mode = Mode.WRITE)
+    @Description("CALL algo.jaccard([{source:id, targets:[ids]}], {similarityCutoff:-1,degreeCutoff:0}) " +
+            "YIELD percentile50, percentile75, percentile90, percentile99, percentile999, percentile100 - computes jaccard similarities")
+    public Stream<SimilaritySummaryResult> jaccard(
+            @Name(value = "data", defaultValue = "null") List<Map<String, Object>> data,
+            @Name(value = "config", defaultValue = "{}") Map<String, Object> config) {
+        ProcedureConfiguration configuration = ProcedureConfiguration.create(config);
+
+        double similarityCutoff = ((Number) config.getOrDefault("similarityCutoff", -1D)).doubleValue();
+        long degreeCutoff = ((Number) config.getOrDefault("degreeCutoff", 0L)).longValue();
+
+        InputData[] ids = fillIds(data, degreeCutoff);
+        long length = ids.length;
+        IntStream sourceIds = IntStream.range(0, (int) length);
+
+        TerminationFlag terminationFlag = TerminationFlag.wrap(transaction);
+
+        int concurrency = configuration.getConcurrency();
+
+        DoubleHistogram histogram = new DoubleHistogram(5);
+
+        Stream<SimilarityResult> stream = jaccardStreamMe(ids, (int) length, terminationFlag, concurrency, similarityCutoff);
+
+        AtomicLong similarityPairs = new AtomicLong();
+
+        if(configuration.isWriteFlag() && similarityCutoff > 0.0) {
+            SimilarityExporter similarityExporter = new SimilarityExporter(api,
+                    configuration.get("relationshipType", "SIMILAR"),
+                    configuration.getWriteProperty("score"));
+
+            Stream<SimilarityResult> similarities = stream.peek(result -> {
```

Comment:
we can probably wrap the lambda into a local function or variable.

Reply:
Yup. Done.